crypto/kyber/p_kyber.c

```diff
@@ -29,6 +29,56 @@ static int pkey_kyber512_keygen(EVP_PKEY_CTX *ctx, EVP_PKEY *pkey) {
   return 1;
 }
 
+static int pkey_kyber512_encapsulate(EVP_PKEY_CTX *ctx,
+                                     uint8_t *ciphertext,
+                                     size_t  *ciphertext_len,
+                                     uint8_t *shared_secret,
+                                     size_t  *shared_secret_len) {
+  *ciphertext_len = KYBER512_CIPHERTEXT_BYTES;
+  *shared_secret_len = KYBER512_SHARED_SECRET_BYTES;
+  if (ciphertext == NULL) { // Caller is getting parameter values.
+    return 1;
+  }
```

Comment:
If the caller is not calling to get parameters should this check that the provided buffers are big enough to hold the resulting cipher text and shared secret before updating the sizes?
```suggestion
  if (ciphertext == NULL) { // Caller is getting parameter values.
    *ciphertext_len = KYBER512_CIPHERTEXT_BYTES;
    *shared_secret_len = KYBER512_SHARED_SECRET_BYTES;
    return 1;
  } else if (*ciphertext_len < KYBER512_CIPHERTEXT_BYTES || shared_secret){
    //error
  } else {
    *ciphertext_len = KYBER512_CIPHERTEXT_BYTES;
    *shared_secret
  }
```

Reply:
That's a good question, I'm not sure how was this part of the API designed. I assumed `ciphertext_len` and `shared_secret_len` are output arguments so we can't perform checks on them.

Reply:
Actually, you are right, I'm reading the documentation of the other functions in EVP_PKEY and they all expect `*_len` to have the length of the output buffer. I'll address this.

Reply:
addressed in the latest change.